### PR TITLE
update to generate swagger docs for mounted routers

### PR DESCRIPTION
**Pull request by bradleyhd**
Hi,

I'm not sure if plumber is currently taking unsolicited PRs, but I'm new to both R and plumber and really wanted the automatic swagger doc generation to work with mounted routers. These changes seem to work for me (at least for walking this list of endpoints for the router children of the root router). Open to any/all feedback/criticism and happy to revise.

Thanks!

(I _think_ this addresses https://github.com/trestletech/plumber/issues/219)

### CODE REVIEW

**Comment by schloerke** [comment]
Hi @bradleyhd !

Thanks for the PR!  Good catch!  (Unsolicited PRs are welcome anytime!)

Couple things to address.  
1. Could the for loop be changed to a 
```r
mountedEndpoints <- unlist(lapply(names(self$mounts), function(path) {
...
  if (length(mountEndpoints) == 0) return(NULL)
...
}))
endpoints <- c(endpoints, mountedEndpoints)
```
`NULL` values  with `unlist()` are removed.  When possible, avoid `answer <- c(answer, newItem)` as it does not scale well.

2. Formatting
* The inner logic seems good for calculating `mount_endpoints`.  Could you upgrade the variable names (`i`,`j`) to something more descriptive? (`mountEndpoint`, `mountEndpointEntry` or something similar)
* Could you change the code to use `camelCase` (vs `snake_case`)?

3. We should make the swaggerFile mount calculations recursive.  In the current state, it is missing swaggerFile information for `p3` (`/sub2/sub3`)
```r
pr <- plumber$new()
pr$handle("GET", "/nested/path/here", function(){})
pr$handle("POST", "/nested/path/here", function(){})

stat <- PlumberStatic$new(".")

pr2 <- plumber$new()
pr2$handle("POST", "/something", function(){})
pr2$handle("GET", "/", function(){})

pr3 <- plumber$new()
pr3$handle("POST", "/else", function(){})
pr3$handle("GET", "/", function(){})

pr$mount("/static", stat)
pr2$mount("/sub3", pr3)
pr$mount("/sub2", pr2)

pr$swaggerFile()
```

Thank you in advance,
Barret

**Comment by schloerke** [comment]
Yes, it solves #219

**Comment by bradleyhd** [comment]
Thanks for the pointers/feedback! I've updated to walk routers/mounts recursively and it seems to work on your example and a few others I tried out. However, if you repeatedly call `pr$swaggerFile()`, the paths are repeatedly prepended with their parent paths. I'm not sure what's idiomatic here for preventing this from happening but something should probably be done 😄

**Comment by bradleyhd** [comment]
@schloerke maybe I'm not looking in the right place, but I don't see any review changes. Mind re-adding them?

**Comment by schloerke** [comment]
My fault.. never hit "submit review".  (GitHub review is new to me)

**Comment by bradleyhd** [comment]
No worries at all. I've pushed the test and clone changes, let me know if there's anything else!

File: R/plumber.R [hunk 1]
```diff
@@ -673,6 +675,31 @@ plumber <- R6Class(
...
+      })
+
+      mounts <- router$mounts
+      mountedEndpoints <- unlist(lapply(names(mounts), function(mountPath){
+        mountedSubrouter <- router$mounts[[mountPath]]
+        mountPath <- sub("^[/]", "", mountPath)
+        mountedEndpoints <- private$swaggerFileWalkMountsInternal(mountedSubrouter, paste(parentPath, mountPath, sep="/"))
+      }))
+
+      c(endpoints, list(mountedEndpoints))
```
Comment (alandipert) [inline R/plumber.R]: Is there a reason you call `list` here? My understanding is that at this point `endpoints` is a list and `mountedEndpoints` is a character vector, so the result will definitely be a list.

Comment (bradleyhd) [reply R/plumber.R]: @alandipert I'm a bit fuzzy on the types but I think endpoints is a list of lists and mountedEndpoints without the extra `list` call is just a list. If you remove that call and try to run swagger documentation, you'll get `Error in for (e in fil) { : invalid for() loop sequence`. When I print out the result of calling `c` above without modification, I get entries like
```
[[1]]`$__no-preempt__14`
<PlumberEndpoint>
...
```

Comment (alandipert) [reply R/plumber.R]: Ah, that makes sense, thanks for clarifying.

**Comment by schloerke** [comment]
Merged in #280 

Thank you @bradleyhd!!

**Comment by turiaso** [comment]
Hi, a quick question. Do you upload the version with this feature 0.4.6.9000 (based on versión downloaded by install_github) to CRAN repositories. Same case on 0.5.0 version, based on changelogs. I understand that both versions are more similars, one is a snapshot another is a release.
Regards!

**Pull request closed without merging**